packages/@lwc/engine-core/src/libs/signal/__tests__/index.spec.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { Signal } from './signal';
+
+describe('signal protocol', () => {
```

Comment:
I created these tests as a basic sanity check for behavior we're expecting from a signals implementation.

The idea is for anyone planning to implement the interface to understand how the framework expects it to behave.